src/interceptors/fetch/index.ts

```diff
@@ -20,25 +20,25 @@ export const interceptFetch: Interceptor = (observer, resolver) => {
   debug('replacing "window.fetch"...')
 
   window.fetch = async (input, init) => {
-    const ref = new Request(input, init)
+    const request = new Request(input, init)
     const url = typeof input === 'string' ? input : input.url
```

Comment:
How about we parse with `request`?

Reply:
I don't think that's a good idea. We should keep the actual logic as close to the original request/options as possible, those being `input` and `init` the consumer provided when constructing fetch. Can we guarantee that the original options will be 100% the same after we provided them to the `Request` class? I think that's how `fetch` works internally, but it's hard to assume that, as we can't see its source code.

Reply:
It is 100% the same according to [`fetch()` specification](https://fetch.spec.whatwg.org/#dom-global-fetch). You can see it uses the `Request` constructor internally, with no related modifications in the other steps.

Reply:
I think it quite a common thing to follow the standard step by step when developing a polyfill / plugin that simulates the standard behaviour. And it's rare to see an implementation (browser in our context) uses its own way. So when the standard states it, it should be safe to rely on -)

---

https://github.com/mswjs/interceptors/blob/549d9606db7674468c87ddf4ecabd427eb133a97/src/interceptors/fetch/index.ts#L31

Also, we should use `document.baseURI` in above if insist on parsing it on our own (without `Request`). A advantage I can think of is to satisfy users that expect to see what was directly passed to `fetch` in

https://github.com/mswjs/interceptors/blob/549d9606db7674468c87ddf4ecabd427eb133a97/src/interceptors/fetch/index.ts#L27

I have no preference.

Reply:
I'd like to discuss it as a separate issue if you don't mind. It makes such decisions more traceable. Thanks.